From 6df4ffc36a8c29b5d9ddfc35a92bf422d2b42d09 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 9 Mar 2022 18:35:39 -0300
Subject: [PATCH 020/163] Fix ununsed fstatat64_time64_statx

It is only called for legacy ABIs.

(cherry picked from commit c7f05bd5342517f3f751e6ea8dec1916b80bee8a)
---
 sysdeps/unix/sysv/linux/fstatat64.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/fstatat64.c b/sysdeps/unix/sysv/linux/fstatat64.c
index a82d3dd4df..2ab914380d 100644
--- a/sysdeps/unix/sysv/linux/fstatat64.c
+++ b/sysdeps/unix/sysv/linux/fstatat64.c
@@ -40,6 +40,11 @@ _Static_assert (sizeof (__blkcnt_t) == sizeof (__blkcnt64_t),
                 "__blkcnt_t and __blkcnt64_t must match");
 #endif
 
+#if (__WORDSIZE == 32 \
+     && (!defined __SYSCALL_WORDSIZE || __SYSCALL_WORDSIZE == 32)) \
+     || defined STAT_HAS_TIME32
+# define FSTATAT_USE_STATX 1
+
 static inline int
 fstatat64_time64_statx (int fd, const char *file, struct __stat64_t64 *buf,
 			int flag)
@@ -73,11 +78,6 @@ fstatat64_time64_statx (int fd, const char *file, struct __stat64_t64 *buf,
 
   return r;
 }
-
-#if (__WORDSIZE == 32 \
-     && (!defined __SYSCALL_WORDSIZE || __SYSCALL_WORDSIZE == 32)) \
-     || defined STAT_HAS_TIME32
-# define FSTATAT_USE_STATX 1
 #else
 # define FSTATAT_USE_STATX 0
 #endif
-- 
2.37.2

